### PR TITLE
Network fields schema validation

### LINKED ISSUE #2703: strict validation of `networks` section in the config schema
Currently `#/definitions/network` accepts any fields. We should be a little more strict about what fields are valid in the networks section.

### PR DESCRIPTION
**Pull request by dbonev**
Fixes #2703 

Signed-off-by: Dimitar Bonev <dimitar.bonev@gmail.com>

### CODE REVIEW

File: tests/acceptance/cli_test.py [hunk 1]
```diff
@@ -536,6 +536,16 @@ def test_up_external_networks(self):
...
 
     @v2_only()
+    def test_networks_schema_validation(self):
+        filename = 'invalid-network-fields.yml'
+
+        self.base_dir = 'tests/fixtures/networks'
+        self._project = get_project(self.base_dir, [filename])
+
+        result = self.dispatch(['-f', filename, 'up', '-d'], returncode=1)
+        assert 'Additional properties are not allowed' in result.stderr
```
Comment (dnephin) [inline tests/acceptance/cli_test.py]: We test the rest of this validation in `tests/unit/config/config_test.py`, I think we should move this test there as well.  You can inline the `invalid-network-fields.yml` config in the test for the unit test as well.

File: compose/config/fields_schema_v2.json [hunk 2]
```diff
@@ -41,7 +41,24 @@
...
+        },
+        "external": {
+          "type": ["boolean", "object"],
+          "properties": {
+            "name": {"type": "string"}
+          },
+          "additionalProperties": false
+        }
+      },
+      "additionalProperties": false
```
Comment (dnephin) [inline compose/config/fields_schema_v2.json]: I think this looks like a good start, but we also support the `ipam` field (https://github.com/docker/compose/blob/master/docs/compose-file.md#network-configuration-reference)

**Comment by dnephin** [comment]
cc @aanand

**Comment by aanand** [comment]
Hey, thanks! After @dnephin's comments are addressed, I think it'll be good to go.

**Comment by dbonev** [comment]
Thanks @dnephin, @aanand. Added ipam and moved testing to the unit test.
As always -- thx for the feedback.

**Comment by dnephin** [comment]
LGTM

**Comment by aanand** [comment]
LGTM

**Comment by aanand** [comment]
Cherry-picked into #2753

**Pull request merged**